ts/features/bonus/cgn/components/detail/eyca/EycaStatusDetailsComponent.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from "react";
+import { Badge, View } from "native-base";
+import { StyleSheet } from "react-native";
+import { EycaCardActivated } from "../../../../../../../definitions/cgn/EycaCardActivated";
+import { H4 } from "../../../../../../components/core/typography/H4";
+import I18n from "../../../../../../i18n";
+import IconFont from "../../../../../../components/ui/IconFont";
+import { IOColors } from "../../../../../../components/core/variables/IOColors";
+import { IOStyles } from "../../../../../../components/core/variables/IOStyles";
+import { Monospace } from "../../../../../../components/core/typography/Monospace";
+import { EycaCardExpired } from "../../../../../../../definitions/cgn/EycaCardExpired";
+import { EycaCardRevoked } from "../../../../../../../definitions/cgn/EycaCardRevoked";
+import { H5 } from "../../../../../../components/core/typography/H5";
+import { localeDateFormat } from "../../../../../../utils/locale";
+import ButtonDefaultOpacity from "../../../../../../components/ButtonDefaultOpacity";
+import { Label } from "../../../../../../components/core/typography/Label";
+import { clipboardSetStringWithFeedback } from "../../../../../../utils/clipboard";
+
+type Props = {
+  eycaCard: EycaCardActivated | EycaCardExpired | EycaCardRevoked;
+};
+
+const styles = StyleSheet.create({
+  rowBlock: {
+    flexDirection: "row"
+  },
+  spaced: {
+    justifyContent: "space-between"
+  },
+  statusBadgeActive: {
+    height: 20,
+    marginTop: 2,
+    backgroundColor: IOColors.aqua
+  },
+  statusBadgeExpired: {
+    height: 20,
+    marginTop: 2,
+    backgroundColor: IOColors.bluegrey
+  },
+  statusBadgeText: { paddingHorizontal: 8 }
+});
+
+const ICON_SIZE = 24;
+
+const EycaStatusDetailsComponent: React.FunctionComponent<Props> = (
+  props: Props
+) => {
+  const badgeByStatus = () => {
+    switch (props.eycaCard.status) {
+      case "ACTIVATED":
+        return (
+          <Badge style={[styles.statusBadgeActive]}>
```

Comment:
Could it be a good refactoring to avoid code replication?

```typescript
const EycaStatusBadge = (
  text: string,
  badgeStyle: StyleProp<ViewStyle>,
  textColor: "bluegreyDark" | "white"
) => (
  <Badge style={badgeStyle}>
    <H5 weight={"SemiBold"} color={textColor} style={styles.statusBadgeText}>
      {text}
    </H5>
  </Badge>
);
```

Reply:
Fixed in 059777b